motion/README.md

```diff
@@ -0,0 +1,122 @@
+```
+Please note that any changes you make in the container are not persistent.
+
+Note: a Dockerfile as well as the provided examples are also contained in the [original MOTION repo](https://encrypto.de/code/MOTION) and very similar instructions can be used 
+to build a Docker container from the original repo directly. See the [MOTION README](https://github.com/encryptogroup/MOTION/blob/master/README.md) for more details. 
+
+## Architecture
+
+MOTION is designed as a C++ library. It defines several classes that
+implement the MPC protocols. It is written in an object-oriented way. 
+Secure data is currently limited to unsigned C integer types.
+Support for floating and fixed-point operations is under active development. 
+MOTION supports SIMD operations for efficient parallelization.
+MOTION is significantly more memory and running time efficient when using SIMD 
+and needs slightly less communication due to better packing of messages, so we strongly encourage to use it.
+
+### Format of examples
+
+Most examples have a similar format: 
+- `<example>_main.cpp` read inputs and set parameters
+- `common/<example>.cpp` create MPC objects, define circuit, and execute the
+  computation
+
+### Circuit representation 
+
+There is no explicit circuit object in MOTION.
+The circuit is defined as a collection of `Shares` (containing primitive `Wire` objects) that are manipulated by `Gate` objects.
+Each `Wire` contains `number_of_simd` data values, e.g., 10 integer values in arithmetic GMW, and computing a sum of two such values, `x` and `y`, would result into
+10 additions, one for each `x[i]`, `y[i]`.
+There is also currently no notion of circuit layers in MOTION.
+In a nutshell, all `Gates` have `Shares` as both inputs and outputs.
+The `Gates` wait for their input `Shares` to become "ready" to start their own evaluation.
+After their evaluation, `Gates` set their output wires and mark them "ready" which gives the green light for the waiting gates to start their evaluation.
+This procedure is done completely asynchronously using stackfull coroutines, namely `Boost.Fibers`. 
+
+### Communication serialization
+
+The communication in MOTION is serialized completely using the efficient `flatbuffers` serialization library.
+
+### Benchmarks
+
+We provide `Statistics` classes for analyzing communication and computation costs.
+They have APIs to convert the collected statistics to either a human-readable format, e.g.,
+to print them into the terminal, or to a machine-readable format, namely JSON, using `Boost.JSON`. 
+
+## Running examples
+We provide all three examples in a [tutorial directory in MOTION](https://github.com/encryptogroup/MOTION/blob/master/src/examples/tutorial), 
+so they are not included here.
+To run our examples, you'll need inputs to the circuit. We've written an input
+generation script for `mult3` and other tutorial examples that allows to quickly test the examples. 
+It will tell you what the expected result is.
```

Comment:
Could you please copy these examples into this repository? This would ensure that all examples are in one central place